client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -185,10 +192,25 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        manageSubscriptions(config);
         Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
         synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
     }
 
+    private void manageSubscriptions(ReaderGroupConfig config) {
+        ReaderGroupConfig oldConfig = getReaderGroupConfig();
+        Set<Stream> newStreams = config.isSubscriberForRetention() ? config.getStartingStreamCuts().keySet() : Collections.emptySet();
+        Set<Stream> oldStreams = oldConfig.isSubscriberForRetention() ? oldConfig.getStartingStreamCuts().keySet() : Collections.emptySet();
+
+        Set<Stream> streamsToSub = Sets.difference(newStreams, oldStreams);
```

Comment:
nit: this can be used too... https://github.com/pravega/pravega/blob/c9dac22c506f6d51b08fbd60b76095f40bb5c308/common/src/main/java/io/pravega/common/util/CollectionHelpers.java#L95

Reply:
Using `filterOut` instead.